pandas/core/groupby/generic.py

```diff
@@ -1056,9 +1058,47 @@ def _cython_agg_blocks(
             result = no_result
             locs = block.mgr_locs.as_array
             try:
-                result, _ = self.grouper.aggregate(
-                    block.values, how, axis=1, min_count=min_count
+
+                cat_method_blacklist = (
+                    "add",
+                    "median",
+                    "prod",
+                    "sem",
+                    "cumsum",
+                    "sum",
+                    "cummin",
+                    "mean",
+                    "max",
+                    "skew",
+                    "cumprod",
+                    "cummax",
+                    "rank",
+                    "pct_change",
+                    "min",
+                    "var",
+                    "mad",
+                    "describe",
+                    "std",
+                    "quantile",
                 )
+                if (
+                    is_categorical_dtype(block.values)
+                    and how not in cat_method_blacklist
+                ):
+                    # perf improvement: aggregate categorical codes with fast cython agg
+                    # for select `how` operations
+                    result_codes, _ = self.grouper.aggregate(
+                        block.values.codes, how, axis=1, min_count=min_count
+                    )
+                    result = Categorical.from_codes(
+                        result_codes,
+                        categories=block.values.categories,
+                        ordered=block.values.ordered,
+                    )
```

Comment:
```suggestion
                        dtype=block.values.dtype
                    )
```

Reply:
I actually tried this and found that it raises a ValueError [here](https://github.com/pandas-dev/pandas/blob/1c0cc62e3/pandas/core/dtypes/dtypes.py#L350): Cannot specify `categories` or `ordered` together with `dtype`